livekit/src/room/participant/mod.rs

```
@@ -248,8 +248,21 @@ pub(super) fn add_publication(
     publication.on_muted({
         let events = inner.events.clone();
         let participant = participant.clone();
+        let rtc_engine = inner.rtc_engine.clone();
         move |publication| {
             if let Some(cb) = events.track_muted.lock().as_ref() {
+                if !publication.is_remote() {
+                    let rtc_engine = rtc_engine.clone();
+                    let publication_cloned = publication.clone();
+                    livekit_runtime::spawn(async move {
+                        let _ = rtc_engine
```

I don't think we should ignore the error here

---

Same as the other PR. We should log instead of returning inside the tokio task